src/VaultV2.sol

```diff
@@ -68,6 +68,8 @@ import {ISharesGate, IReceiveAssetsGate, ISendAssetsGate} from "./interfaces/IGa
 /// the same ids.
 /// @dev If allocations underestimate the actual assets, some assets might be lost because deallocating is impossible if
 /// the allocation is zero.
+/// @dev On allocation or deallocation, the underlying position of an adapter may become underestimated, for instance due to roundings or entry/exit fees.
+/// @dev This understimate is only resolved in the next transaction, and so it should be small to avoid excessive fees and withdrawals at unfair share prices.
```

Comment:
should we say that the adapter might not statically prevent it, but that it should be used in ways that prevent it (like for re-entrance)?

Reply:
I don't find a good phrasing for this, do you have an idea? The issue is that it addresses possible future adapters (some might prevent it statically, some might not).